lib/storage/WebStorage.js

```diff
@@ -0,0 +1,49 @@
+import _ from 'underscore';
+import Storage from './providers/LocalForage';
+
+const SYNC_ONYX = 'SYNC_ONYX';
+
+/**
+ * Raise an event thorough `localStorage` to let other tabs know a value changed
+ * @param {String} onyxKey
+ */
+function raiseStorageSyncEvent(onyxKey) {
+    global.localStorage.setItem(SYNC_ONYX, onyxKey);
+    global.localStorage.removeItem(SYNC_ONYX, onyxKey);
+}
+
+const webStorage = {
+    ...Storage,
+
+    /**
+     * Storage synchronization mechanism keeping all opened tabs in sync
+     * @param {function(key: String, data: *)} onStorageKeyChanged
+     */
+    keepInstancesSync(onStorageKeyChanged) {
+        // Override set, remove and clear to raise storage events that we intercept in other tabs
+        this.setItem = (key, value) => Storage.setItem(key, value)
+            .then(() => raiseStorageSyncEvent(key));
+
+        this.removeItem = key => Storage.removeItem(key)
+            .then(() => raiseStorageSyncEvent(key));
+
+        // If we just call Storage.clear other tabs will have no idea which keys were available previously
+        // so that they can call keysChanged for them. That's why we iterate and remove keys one by one
+        this.clear = () => Storage.getAllKeys()
+            .then(keys => _.forEach(keys, key => this.removeItem(key)));
```

Comment:
Just a thought, but curious to see how this performs in the case of a sign out. There is some temporary code added [here](https://github.com/Expensify/App/blob/main/src/libs/SignoutManager.js) to ensure the sign outs are synced and I think we can remove it if this works how I think it does.

Reply:
I've altered the logic slightly 
1. WebStorage: wait for all remove calls to finish and then resolve clear - use `Promise.all`
2. Bug fix: on the new tab we were only `keysChanged` - which kept the old value in `cache`. Added an update to set the new value in cache and call `keysChanged` as well 

> There is some temporary code added [here](https://github.com/Expensify/App/blob/main/src/libs/SignoutManager.js) to ensure the sign outs are synced and I think we can remove it if this works how I think it does.

That `true` I've tested this and sign out works correctly with or without the `SignoutManager`

https://user-images.githubusercontent.com/12156624/140195176-0a584ac7-648c-45e1-8dc5-3b0f7a4ef15c.mp4